modules/juce_core/native/juce_Network_linux.cpp

```diff
@@ -525,7 +525,14 @@ class WebInputStream::Pimpl
         if (userHeaders.isNotEmpty())
             header << "\r\n" << userHeaders;
 
-        header << "\r\n\r\n";
+        if(header.toString().endsWith("\r\n"))

```

Comment:
I am sure there's a better to detect `\r\n` than creating a string copy here.

Reply:
I have added another commit which uses a reference to a variable.

Reply:
What I meant was the `toString` call, which I think creates a string copy. The actual string literals were fine. 
It should not matter much as headers should be small.